tensorflow_datasets/image/cartoonset.py

```diff
@@ -0,0 +1,142 @@
+    """
+    if data not in _DATA_OPTIONS:
+      raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+
+    super(CartoonsetConfig, self).__init__(**kwargs)
+    self.data = data
+
+
+class Cartoonset(tfds.core.GeneratorBasedBuilder):
+  """CartoonSet is a collection of random, 2D cartoon avatar images."""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  BUILDER_CONFIGS = [
+      CartoonsetConfig(
+          name=config_name,
+          description="A collection of random, 2D cartoon avatar images",
+          version="0.1.0",
+          data=config_name,
+      ) for config_name in _DATA_OPTIONS
+  ]
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict(
+            {"image": tfds.features.Image(shape=_CARTOON_IMAGE_SHAPE),
+             "eye_angle": tfds.features.ClassLabel(num_classes=3),
+             "eye_lashes": tfds.features.ClassLabel(num_classes=2),
+             "eye_lid": tfds.features.ClassLabel(num_classes=2),
+             "chin_length": tfds.features.ClassLabel(num_classes=3),
+             "eyebrow_weight": tfds.features.ClassLabel(num_classes=2),
+             "eyebrow_shape": tfds.features.ClassLabel(num_classes=14),
+             "eyebrow_thickness": tfds.features.ClassLabel(num_classes=4),
+             "face_shape": tfds.features.ClassLabel(num_classes=7),
+             "facial_hair": tfds.features.ClassLabel(num_classes=15),
+             "hair": tfds.features.ClassLabel(num_classes=111),
+             "eye_color": tfds.features.ClassLabel(num_classes=5),
+             "face_color": tfds.features.ClassLabel(num_classes=11),
+             "hair_color": tfds.features.ClassLabel(num_classes=10),
+             "glasses": tfds.features.ClassLabel(num_classes=12),
+             "glasses_color": tfds.features.ClassLabel(num_classes=7),
+             "eye_slant": tfds.features.ClassLabel(num_classes=3),
+             "eyebrow_width": tfds.features.ClassLabel(num_classes=3),
+             "eye_eyebrow_distance": tfds.features.ClassLabel(num_classes=3)}),
+        supervised_keys=None,
+        # Homepage of the dataset for documentation
+        urls=["https://google.github.io/cartoonset/"],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    # There is no predefined train/val/test split for this dataset.
+    path = dl_manager.manual_dir
+    if not tf.io.gfile.exists(path):
+      msg = 'You must download the dataset files manually and place them in: '
+      msg += ', '.join([path])
+      raise AssertionError(msg)
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=2,
+            gen_kwargs={
+                "filepath": path
+            })
+    ]
+
+  def _generate_examples(self, filepath):
+    """Yields examples."""
+    for path, _, files in tf.io.gfile.walk(filepath):
```

Comment:
replace this line with `for path, _, files in tf.io.gfile.walk(filepath+'/'+self.builder_config.name):` otherwise if 10k and 100k both are present, then it will generate (10k + 100k), i.e. it mixes all the images from the configs.